mypy/test/teststubtest.py

```diff
@@ -224,6 +227,65 @@ def test_coroutines(self) -> Iterator[Case]:
         yield Case(
             stub="async def bingo() -> int: ...", runtime="async def bingo(): return 5", error=None
         )
+        # Be more permissive if it's an abstractmethod in the stub...
+        yield Case(
+            stub="""
+            from abc import abstractmethod
+            from typing import Awaitable
+            class Bar:
+                @abstractmethod
+                def abstract(self) -> Awaitable[int]: ...
+            """,
+            runtime="""
+            class Bar:
+                async def abstract(self): return 5
+            """,
+            error=None,
+        )
+        # ...and/or an abstractmethod at runtime...
+        yield Case(
+            stub="""
+            from abc import abstractmethod
+            from typing import Generator, Any, Coroutine, Awaitable
+            class _CustomAwaitable:
+                def __await__(self) -> Generator[Any, Any, Any]: ...
+            class _CustomAwaitable2(Awaitable[Any]): ...
+            class Foo:
+                @abstractmethod
+                def abstract(self) -> _CustomAwaitable: ...
```

Comment:
Why do we want to allow this? Doesn't seem particularly useful.

Reply:
I can't remember, and I'm not sure I care anymore. I guess I'll just close this PR :)